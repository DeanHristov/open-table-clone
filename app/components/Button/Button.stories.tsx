import * as React from "react";
import { Meta, Story } from "@storybook/react";
import Button, { IButtonProps } from "./Button";
import { ArrowTrendingUpIcon } from "@heroicons/react/24/solid";

export default {
  title: "Components/Button",
  component: Button,
  parameters: {
    actions: {
      handles: ["click"],
    },
  },
} as Meta;

export const Default: Story<IButtonProps> = (args) => <Button {...args} />;
export const WithIcon: Story<IButtonProps> = (args) => <Button {...args} />;

Default.args = {
  title: "Click me!",
  medium: false,
  blue: false,
  red: false,
  disabled: false,
  white: false,
};

WithIcon.args = {
  title: "Click me!",
  icon: <ArrowTrendingUpIcon className="h-6 w-6 text-black inline-block" />,
  medium: false,
  blue: false,
  red: false,
  disabled: false,
  white: false,
};
