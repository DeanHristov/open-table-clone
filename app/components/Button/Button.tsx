"use client";

import { FC, ReactNode } from "react";
import classNames from "classnames";

export interface IButtonProps {
  title: string;
  icon?: ReactNode;
  medium?: boolean;
  red?: boolean;
  blue?: boolean;
  white?: boolean;
  disabled?: boolean;
  onClick?: () => void;
}

const Button: FC<IButtonProps> = ({
  title,
  icon,
  medium = false,
  blue = false,
  red = false,
  white = false,
  disabled = false,
  onClick = () => {},
}) => {
  const baseStyles = [
    "rounded w-full px-4 py-2 text-center text-sm",
    "transition-all duration-300",
    "hover:shadow-md cursor-pointer",
    "disabled:cursor-not-allowed",
  ];
  const styles = classNames(baseStyles, {
    "bg-shades-100 text-white hover:bg-shades-200": blue,
    "bg-white border border-red-600 text-red-500": white,
    "bg-red-600 text-white hover:bg-red-700": red,
    "bg-white hover:bg-gray-50 border text-black": !blue && !white && !red,
    "h-12 min-w-[132px]": medium,
  });

  // TODO Fix it -> Unhandled Runtime Error
  // Error: Hydration failed because the initial UI does not match what was rendered on the server.

  return (
    <div className="p-1 h-auto">
      <button disabled={disabled} onClick={onClick} className={styles}>
        {icon} {title}
      </button>
    </div>
  );
};

export default Button;
