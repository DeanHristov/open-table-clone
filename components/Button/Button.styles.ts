import styled from "styled-components";

export const StyledButton = styled.button`
  width: 30rem;
  height: 3rem;
  border: 0.1rem solid red;
  padding: 0.5rem;
`;
