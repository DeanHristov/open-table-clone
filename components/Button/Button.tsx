import { FC } from "react";

export interface IButton {
  title: string;
}

const Button: FC<IButton> = ({ title }) => {
  return <button>{title}</button>;
};

export default Button;
