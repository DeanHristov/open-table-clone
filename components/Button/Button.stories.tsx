import * as React from "react";
import {Meta, Story} from "@storybook/react";
import Button, {IButton} from "./Button"

export default {
    title: "Components/Button",
    component: Button,
    parameters: {
        actions: {
            handles: ["click"],
        },
    },
    argTypes: {},
} as Meta;

export const Default: Story<IButton> = (args) => <Button {...args} />;

Default.args = {
    title: "Click me!"
};
