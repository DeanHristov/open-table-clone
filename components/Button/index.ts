import Button, {IButton} from "./Button";

export type {IButton}
export default Button;