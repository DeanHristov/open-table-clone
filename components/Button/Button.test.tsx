import {render} from "@testing-library/react";
import Button from "./Button";

describe("Component: <Button {...} />", () => {
    it("Should be able to render default state", () => {
        const {container} = render(
            <Button title="click me"/>
        );

        expect(container).toMatchSnapshot();
    })
})