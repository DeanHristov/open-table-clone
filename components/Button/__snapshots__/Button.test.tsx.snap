// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Button {...} /> Should be able to render default state 1`] = `
<div>
  <button>
    click me
  </button>
</div>
`;
